packages/solid-loaders/src/index.tsx

```diff
@@ -0,0 +1,164 @@
+import {
+  Loader,
+  LoaderClient,
+  LoaderClientOptions,
+  LoaderClientStore,
+  LoaderInstance,
+  LoaderInstanceByKey,
+  LoaderStore,
+  RegisteredLoaders,
+  StrictLoaderInstance,
+  VariablesOptions,
+} from '@tanstack/loaders'
+import * as Solid from 'solid-js'
+import * as SolidStore from 'solid-js/store'
+import invariant from 'tiny-invariant'
+
+import { useStore } from '@tanstack/solid-store'
+
+export * from '@tanstack/loaders'
+
+//
+
+export type NoInfer<T> = [T][T extends any ? 0 : never]
+
+const loaderClientContext = Solid.createContext<LoaderClient<any>>(null as any)
+
+export function LoaderClientProvider(
+  props: {
+    loaderClient: LoaderClient<any>
+    children: any
+  } & Pick<LoaderClientOptions<any>, 'defaultGcMaxAge' | 'defaultMaxAge'>,
+) {
+  const [coreProps, optionProps] = Solid.splitProps(props, [
+    'loaderClient',
+    'children',
+  ])
+
+  const [loaderClient, setLoaderClient] = SolidStore.createStore(
+    coreProps.loaderClient,
+  )
+
+  Solid.createEffect(
+    Solid.on(
+      () => [coreProps.loaderClient.options, optionProps],
+      ([loaderOptions, optionProps]) => {
+        setLoaderClient('options', (o) => ({
+          ...o,
+          ...loaderOptions,
+          ...optionProps,
+        }))
+      },
+    ),
+  )
+
+  return (
+    <loaderClientContext.Provider value={loaderClient}>
+      {coreProps.children}
+    </loaderClientContext.Provider>
+  )
+}
+
+type StrictLoaderInstance_<TLoaderInstance> =
```

Comment:
Decided to switch this in the core so `data` is always defined, but then in the adapter, we now have `NullableLoaderInstance_` which references the core `NullableLoaderInstance`.